hack/import-restrictions.yaml

```diff
@@ -152,3 +152,9 @@
   - k8s.io/api
   - k8s.io/apimachinery
   - k8s.io/cluster-bootstrap
+
+- baseImportPath: "./vendor/k8s.io/cloud-provider/"
+  allowedImports:
+  - k8s.io/api
+  - k8s.io/apimachinery
+  - k8s.io/client-go
```

Comment:
no self-imports? Can we verify whether the self-import rule is actually needed?

Reply:
> no self-imports?

Yes, cloud-provider doesn't have any self imports.

> Can we verify whether the self-import rule is actually needed?

Yes, it is necessary. Verified by commenting out the self-import rule for cluster-bootstrap:

```yaml
- baseImportPath: "./vendor/k8s.io/cluster-bootstrap/"
  allowedImports:
  - k8s.io/api
  - k8s.io/apimachinery
  # - k8s.io/cluster-bootstrap
```

`hack/verify-imports.sh` fails with:

```
2018/10/09 14:26:33 Inspecting imports under ./vendor/k8s.io/cluster-bootstrap/...
2018/10/09 14:26:33 - validating imports for 2 packages in the tree
2018/10/09 14:26:33 -- found forbidden imports for k8s.io/kubernetes/vendor/k8s.io/cluster-bootstrap/token/util:
2018/10/09 14:26:33 --- k8s.io/cluster-bootstrap/token/api
2018/10/09 14:26:33 - FAIL
```

Which makes sense because this file in cluster-bootstrap has a self-import:

https://github.com/kubernetes/kubernetes/blob/4ba5db087a2d7af5c04c99cb99e14ae74000fcbf/staging/src/k8s.io/cluster-bootstrap/token/util/helpers.go#L27

Reply:
Just to be clear, this is just the cloud provider interface and registry. Plenty of cloud provider implementations refer to both themselves and the interface, but this code is clear of self-reference.